### ISSUE TITLE
Upgrade scraping to handle formatting poetry and non-standard text properly

### ISSUE THREAD

**Issue by pmfischer80** [UnimatrixZeroOne/BibleBot#34]
How it should look: https://i.imgur.com/nnPK9ZL.png

How it looks now:  https://i.imgur.com/FAbQeUM.png

The scraping on this is much more complex.  I've reviewed to some degree and would be willing to help with examining the html to figure out what has to be done

**Comment by vypr** [comment]
I'm unsure how I can go about doing this, in terms of the text formatting. I am able to have the bot tell when something is poetry or not, but having the text formatted properly will be an interesting task. I'll see how this goes.

**Comment by vypr** [comment]
Maybe through the usage and reading of the "indent-1" and "indent-1-breaks" classes in there, maybe we could have the bot detect it and add a certain amount of spaces or tabs accordingly.

**Comment by pmfischer80** [comment]
When I get a chance I will skim the html, they use different classes in
different places (and different versions) so we'll need to identify all the
useful ones and come up with a method to handle that.... I can help with
that, coming up with a method (algorithm) to do this.  I know it's
definitely doable I just couldn't work out the code in node.js and I didn't
want to reinvent the wheel in a language I know better.

**Comment by vypr** [comment]
Well, give me the code that you would use in your language and I can transpile. Even if it's pseudo-code I can try something out.

**Comment by pmfischer80** [comment]
Ok, I'm going to break this down into two parts:


*Part 1 will be Paragraph Breaks Part 2 will be line breaks in poetry and
indenting*

**NOTE**  I'm only checking the ESV for now, I would recommend implementing
either a command that would allow you to turn on this new formatting or use
the old (or implement something in the code that allows different
formatting sets to different versions because I'm not sure this is going to
be standard across all versions.  If we can make progress on the ESV then
we can move on to other popular versions and eventually make sure we have
rules that account for everything.

*Part 1: Paragraph breaks*

Analysis:

   1. Not poetry paragraphs (from here on out called Normal lines) are
   simple <p> tags with a class name that doesn't matter to us (sometimes it's
   the chapter name and sometimes it's "Chapter 1", sometimes it's
   "first-line-none top-1".
   2. Poetry paragraphs are in <div> tags with a class name "poetry" or
   "poetry top-1"
   3. Headings are in <h3> tags


All three of these types of tags sit at the same level nested under a div
that has a class name based on the version and formatting style.

Recommendation:
I think the simplest solution would be to put in a paragraph break (which
would just be two line breaks for discord formatting) before and after each
of these three types of formats and then strip the extra line breaks from
the output (including all line breaks at the beginning or end of the output)

*Part 2: Line breaks and indenting*

Analysis:
Inside of the "poetry" and "poetry top-1" <div> tags is a <p> tag with a
class name "line" and inside that tag are <span> tags.  Each <span> tag
contains individual lines of poetry and if the line needs to be indented
the span has a class name of "indent-1".

Recommendation:
If the text is inside one of these spans tags that are inside a "poetry" or
"poetry top-1" indent it 4 spaces minus the length of the verse number and
add a line break after the text.  And if it's class name is "indent-1" then
add an additional 2 spaces to the beginning of the line.

**Comment by vypr** [comment]
Part 1 - pseudocode:
```
if poetry div exists:
   content = "\n\n" + content + "\n\n"
```

tl;dr this will provide the right amount of separation between the poetry divs. for headings **in** poetry, i'll have to see the html myself to make sure i can select everything properly

Part 2 - pseudocode:

```
if a poetry div exists:
    var span = span inside div
    if span has class indent-1:
       span = "  " + span
   span = "    " + span;
```

tl;dr put 4 spaces before poetry span, add two more if it has indent-1
i need to take a look at how verse numbers appear, i'd like for the result to be something like:

```
[1]    First line of poetry.
            Second line of poetry.
[2]    Third line of poetry.
            Fourth line of poetry.
```

so on

**Comment by pmfischer80** [comment]
I was doing some testing today and Biblebot doesn't do any paragraph breaks currently, so adding paragraph breaks for everything, not just poetry should be combined with this issue.

**Comment by pmfischer80** [comment]
Here's the main layout from a chapter in the ESV that has a lot of poetry and none poetry lines:
https://i.imgur.com/XFdh0Re.png

The main div everything is in has a class name that starts with "version-" if you can grab that div, everything else you need is inside it.

Headings are <span>'s inside of <h3>'s
https://i.imgur.com/t08b65l.png

Non-poetry is <span>'s inside of <p>'s
https://i.imgur.com/3P77aol.png

Poetry is <span>'s inside of <p>'s inside of <div>'s
https://i.imgur.com/nrhsgMo.png

I'll try to help update your pseudo-code for part 1 so that it encompasses everything we're trying to do here.
